src/pudl/transform/eiaaeo.py

```diff
@@ -514,6 +514,105 @@ def core_eiaaeo__yearly_projected_generation_in_end_use_sectors_by_fuel_type(
+    information about what is included or excluded in each usage type for each sector
+    in the footnotes of the EIA's online AEO data browser:
+
+      https://www.eia.gov/outlooks/aeo/data/browser/#/?id=2-AEO2023
+
+    The data browser also gives some visibility into the tricky system of subtotals
+    within the Energy Use series. To identify and map subtotal usage types, look for
+    the following features in the data browser display: subtotal series are displayed
+    indented, and include all lines above them which are one level out, up to the
+    next indented line. Delivered Energy and Total are special cases which include
+    those plus all subtotals above. In this way, "Delivered Energy" includes
+    purchased electricity, renewable energy, and an array of fuels based on sector,
+    and explicitly excludes electricity-related losses.
+
+    AEO Energy Use figures are variously referred to as delivered energy, energy
+    consumption, energy use, and energy demand, depending on which usage types are
+    being discussed, and which org and which document is describing them. In PUDL we
+    say energy use or energy consumption.
+    """
+    sanitized = filter_enrich_sanitize(
+        raw_df=raw_eiaaeo__energy_consumption_by_sector_and_source,
+        relevant_series_names=(
+            "Energy Use : Commercial",
+            "Energy Use : Electric Power",
+            "Energy Use : Industrial",
+            "Energy Use : Residential",
+            "Energy Use : Total",
+            "Energy Use : Transportation",
+            "Energy Use : Unspecified",
+        ),
+    ).rename(columns={"subtopic": "dimension"})
+
+    expected_values = {
+        "topic": {"energy_use"},
+        #         "variable_name": {"delivered_energy"},
+        "units": {"quads"},
+        "dimension": {
+            "commercial",
+            "electric_power",
+            "industrial",
+            "residential",
+            "total",
+            "transportation",
+            "unspecified",
+        },
+    }
+    for column, expected in expected_values.items():
+        assert (actual := set(sanitized[column].unique())) == expected, (
+            f"Unexpected values in {column}: Expected {expected}; found {actual}"
+        )
+
+    # 1 quad = 1 quadrillion btu = 1e9 mmbtu
+    sanitized.loc[sanitized.units == "quads", "value"] *= 1e9
+
+    trimmed = sanitized.drop(
+        columns=[
+            "topic",
+            "units",
+        ]
+    )
+
+    unstacked = (
+        trimmed.set_index(
+            [
+                "report_year",
+                "model_case_eiaaeo",
+                "region",
+                "dimension",
+                "projection_year",
+                "variable_name",
+            ]
+        )
+        .sort_index()
+        .assign(region_type="us_census_division")
+        .reset_index()
+    )
+    unstacked.loc[unstacked.region == "united_states", "region_type"] = "nation"
```

Comment:
we're not technically unstacking here, so that we can keep the table long instead of having a separate partially-empty column for each use type.

thoughts on merits of consistency with the other transforms (which do unstack here) vs clarity of variable name?